_posts/2021-09-26-newsletter.md

```diff
@@ -0,0 +1,126 @@
+concept had more general value beyond only LSP.
+The development of `vim.diagnostic` diagnostic framework from concept, to
+interface design, to delivery was a matter of weeks, thanks to @gpanders.
+
+The `vim.diagnostic` API is available in Nvim master (unreleased 0.6) and we are considering it for 0.5.2 stable release.
+
+Thoug the Nvim RPC API is bound by a sacred
+[contract](https://neovim.io/doc/user/api.html#api-contract), in other ares of Nvim we are now more
+aggressively deprecating and reworking interfaces.  We are fundamentally committed to backwards compatibility for the RPC API and will eventually adopt that commitment in the Lua interface.  Meanwhile, we reserve the option of breaking things in Lua-land. Such breakage is collected and announced in the [Following HEAD](https://github.com/neovim/neovim/issues/14090) thread.
+
+This liberty has allowed us to move 10x faster.
+
+### Lua plugins
+
+@bfredl continues to improve Lua and Vimscript "runtime path" behavior to be intuitive, useful, and fast.
+
+The model of "core framework, external dust cloud" has proven out. nvim-treesitter external project is heavily experimenting and some of its ancillary efforts such as UI "widgets" are finding their way back to core.
+
+nvim-lspconfig saw heavy churn for 1+ years and is now settling down. Some of its utilities such as `:LspInfo` and the "root finder" mechanism are being considered as core Nvim feature.
+
+### Lua config
+
+Nvim supports `init.lua` as user config since 0.5, and in 0.6. we are working on `nvim -l` to support invoking Lua "main" scripts with `nvim`.
+
+## Language Server Protocol (LSP)
+
+The [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) is an
+open, JSON-RPC protocol for communication between code editors and
+language servers, which provide programming language-specific features such as
+completion, hover/tooltips, go to definition, show/go to references, show
+method signatures, rename, refactoring.
+
+Nvim's builtin [`vim.lsp`](https://neovim.io/doc/user/lsp.html) [LSP framework](https://www.youtube.com/watch?v=ArwDgvYEZYk)
+plus [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) quickstart collection has been a major success,
+demonstrating the viability of shipping ambitious ideas in core Nvim using Lua.
+Nvim core developers use it to accelerate their work in Nvim's C and Lua source code.
+
+Work on [semantic highlighting](https://github.com/neovim/neovim/pull/14122) continues.
+
+## Tree-sitter
+
+Work on [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) parser integration continues: performance, bug-fixes.
+
+Also, Neovim itself only provides a (Lua) API for generating and querying the syntax tree using the bundled tree-sitter library; see [`:h treesitter`](https://neovim.io/doc/user/treesitter.html). User-facing features like the ones mentioned above are implemented in plugins like
```

Comment:
(note: unfinished sentence here `implemented in plugins like .....`)
maybe use `XXX` to indicate 'fill-me' areas? :eyes: 
```suggestion
Also, Neovim itself only provides a (Lua) API for generating and querying the syntax tree using the bundled tree-sitter library; see [`:h treesitter`](https://neovim.io/doc/user/treesitter.html). User-facing features like the ones mentioned above are implemented in plugins like XXX and XXX.
```